monkey/agent_plugins/exploiters/snmp/src/plugin.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from functools import partial
+from pprint import pformat
+from typing import Any, Dict, Sequence
+
+# common imports
+from common.event_queue import IAgentEventPublisher
+from common.types import AgentID, Event
+from common.utils.code_utils import del_key
+
+# dependencies to get rid of or internalize
+from infection_monkey.exploit import IAgentBinaryRepository, IAgentOTPProvider
+from infection_monkey.exploit.tools import all_exploitation_ports_are_closed
+from infection_monkey.exploit.tools.http_agent_binary_server import start_agent_binary_server
+from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.network import TCPPortSelector
+from infection_monkey.propagation_credentials_repository import IPropagationCredentialsRepository
+
+logger = logging.getLogger(__name__)
+
+
+class StubSNMPOptions:
+    def __init__(self, *args, **kwargs):
+        pass
+
+
+class StubSNMPExploiter:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def exploit_host(self, *args, **kwargs):
+        pass
+
+
+class StubSNMPExploitClient:
+    def __init__(self, *args, **kwargs):
+        pass
+
+
+SNMP_PORTS = [161]
+
+
+def should_attempt_exploit(host: TargetHost) -> bool:
+    return not all_exploitation_ports_are_closed(host, SNMP_PORTS)
```

Comment:
This isn't going to achieve the desired results, since this function only operates on TCP ports, but SNMP user UDP.

Reply:
Nice catch